src/origyn_nft_reference/metadata.mo

```diff
@@ -1238,10 +1238,51 @@ module {
     };
 
     SB.add(ledger, newTrx);
+    
+    // Call for the annouce function
+    announceTransaction(state, rec, caller, newTrx);
 
     return #ok(newTrx);
   };
 
+  public func announceTransaction(state : Types.State, rec : Types.TransactionRecord, caller : Principal, newTrx : Types.TransactionRecord) : async Types.TransactionRecord {
+        let subscriberId = caller;
+        let eventName = "transaction";
```

Comment:
We need to use universal namespaces that will be unique everywhere.  Our base namespace for the NFT should be "com.origyn.nft.event".  Then each transaction type should have its own namespace based off of the type.  So you'll likely need to do a switch here and handle all of the rec.types.